Docs: Adding a Markdown Style Guide

RobinJoe opened the pull request:
What this PR does / why we need it: This request builds a markdown style guide for writers documenting information about Grafana services for users.
This PR serves the need to make documentation and READMEs more consistent and maintainable.

Which issue(s) this PR fixes: This request fixes the issue grafana#19592

Fixes grafana#19592

Special notes for your reviewer:

@marcusolsson and @oddlittlebird, sorry for the delay. I've set up the style guide, and it is ready for review, or additional sections.

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/master/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the master branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**: This request builds a markdown style guide for writers documenting information about Grafana services for users.
This PR serves the need to make documentation and READMEs more consistent and maintainable.

**Which issue(s) this PR fixes**: This request fixes the issue grafana#19592

<!--

* Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes grafana#19592

**Special notes for your reviewer**:

@marcusolsson and @oddlittlebird, sorry for the delay. I've set up the style guide, and it is ready for review, or additional sections.

RobinJoe committed:
Docs: Adding a Markdown Style Guide

What this PR does / why we need it: This request builds a markdown style guide for writers documenting information about Grafana services for users.
This PR serves the need to make documentation and READMEs more consistent and maintainable.

Which issue(s) this PR fixes: This request fixes the issue grafana#19592

Fixes grafana#19592

Special notes for your reviewer:

@marcusolsson and @oddlittlebird, sorry for the delay. I've set up the style guide, and it is ready for review, or additional sections.

RobinJoe committed:
Docs: Adding a Markdown Style Guide

What this PR does / why we need it: This request builds a markdown style guide for writers documenting information about Grafana services for users.
This PR serves the need to make documentation and READMEs more consistent and maintainable.

Which issue(s) this PR fixes: This request fixes the issue grafana#19592

Fixes grafana#19592

Special notes for your reviewer:

@marcusolsson and @oddlittlebird, sorry for the delay. I've set up the style guide, and it is ready for review, or additional sections.

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=22425) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=22425) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=22425) it.</sub>

RobinJoe committed:
Merge branch 'JR-MDStyleGuide-020220' of github.com:RobinJoe/grafana into JR-MDStyleGuide-020220

RobinJoe committed:
Docs: fixing typos and char repetition

Editing some typos in the text, and removing the repetition of A and B characters that appeared in the file.

oddlittlebird commented on contribute/style-guides/documentation-markdown-guide.md:
```diff
@@ -0,0 +1,147 @@
+
+**Example** Emoji became popular in the **mid-2000's**
+
+## Links and References
```
Per the documentation style guide, all headings should be in sentence case. Initial cap only.

oddlittlebird commented on contribute/style-guides/documentation-markdown-guide.md:
```diff
@@ -0,0 +1,147 @@
+
+\[text to display](www.website.com)
+
+**Example** For more information on including emoji in Github Flavored Markdown, refer to the [webfx page on emoji](https://www.webfx.com/tools/emoji-cheat-sheet/) for a list of emoji.
```
```suggestion
**Example:** For more information on including emoji in Github Flavored Markdown, refer to the [webfx page on emoji](https://www.webfx.com/tools/emoji-cheat-sheet/) for a list of emoji.
```

oddlittlebird commented on contribute/style-guides/documentation-markdown-guide.md:
```diff
@@ -0,0 +1,147 @@
+
+**Example** For more information on including emoji in Github Flavored Markdown, refer to the [webfx page on emoji](https://www.webfx.com/tools/emoji-cheat-sheet/) for a list of emoji.
+
+## Blockquotes
```
Block quotes is two words.

oddlittlebird commented on contribute/style-guides/documentation-markdown-guide.md:
```diff
@@ -0,0 +1,147 @@
+
+## Blockquotes
+
+Include Blockquotes inside text using right facing arrows:
```
```suggestion
Include block quotes inside text using right-facing arrows:
```

oddlittlebird commented on contribute/style-guides/documentation-markdown-guide.md:
````diff
@@ -0,0 +1,147 @@
+```
+
+Write the name of the language after the first set of back tics, no spaces,
+to show specific syntax highlighting - for example "\```javascript" produces the following
````
```suggestion
to show specific syntax highlighting. For example: "\```javascript" produces the following
```

RobinJoe committed:
Merge pull request #1 from grafana/master

Edit- no style guide. A fork update.

RobinJoe committed:
Merge branch 'JR-MDStyleGuide-020220' of github.com:RobinJoe/grafana into JR-MDStyleGuide-020220

RobinJoe committed:
Docs: Changing content following review of new Markdown Style Guide

- adjustment to headings to sentence case.
- adding an introduction
- removing punctuation content
- making suggested word and punctuation fixes

Fixes #19592

RobinJoe commented:
Looking at the latest commit id (https://github.com/grafana/grafana/pull/22425/commits/224ee8aff4a766ea5f5c7f183212bbc53ed6419d) the content does appear to be a PR aimed to merge into Grafana/Grafana repo. I compared it to other commits tagged with pr/external. I made the requested changes in the latest commit if that helps clarify what stage the work is up to.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/grafana/grafana?pullRequest=22425) <br/>All committers have signed the CLA.

oddlittlebird commented:
LGTM! Thanks for contributing this resource!
